packages/bot-skeleton/src/services/api/api-base.js

```diff
@@ -32,10 +34,24 @@ class APIBase {
         if (this.api) this.api.disconnect();
     }
 
+    stopBotIfSocketDisconnected() {
+        if (this.api.connection.readyState === 3) {
+            const { stopBot, clearStat } = DBotStore.instance.run_panel;
+            const { updateTransactionIfSocketDisconnected } = DBotStore.instance.transactions;
+            stopBot();
+            clearStat(true);
+            updateTransactionIfSocketDisconnected();
+        }
+    }
+
     initEventListeners() {
         if (window) {
             window.addEventListener('online', this.reconnectIfNotConnected);
             window.addEventListener('focus', this.reconnectIfNotConnected);
+            //we can even run the bot by opening a new websocket connection here!
+            document.addEventListener('visibilitychange', () => {
+                this.stopBotIfSocketDisconnected();
+            });
```

Comment:
Can we close this listener if bot is not running?

Reply:
done ✅